unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -1793,61 +1810,102 @@ public class PhysicsRemoteFPSAgentController : BaseFPSAgentController {
             return colliders;
         }
 
-        public override void MoveLeft(ServerAction action) {
-            action.moveMagnitude = action.moveMagnitude > 0 ? action.moveMagnitude : gridSize;
+        public virtual void MoveLeft(
+            float? moveMagnitude = null,
+            string objectId = "",
+            float maxAgentsDistance = -1f,
+            bool forceAction = false,
+            bool manualInteract = false,
+            bool allowAgentsToIntersect = false
+        ) {
+            if (!moveMagnitude.HasValue) {
+                moveMagnitude = gridSize;
+            } else if (moveMagnitude < 0f) {
+                throw new InvalidOperationException("moveMagnitude but be > 0");
+            }
+
             actionFinished(moveInDirection(
-                -1 * transform.right * action.moveMagnitude,
-                action.objectId,
-                action.maxAgentsDistance,
-                action.forceAction,
-                action.manualInteract,
-                action.allowAgentsToIntersect ? allAgentColliders() : null
+                direction: -transform.right * moveMagnitude.Value,
+                objectId: objectId,
+                maxDistanceToObject: maxAgentsDistance,
+                forceAction: forceAction,
+                manualInteract: manualInteract,
+                ignoreColliders: allowAgentsToIntersect ? allAgentColliders() : null
             ));
         }
 
-        public override void MoveRight(ServerAction action) {
-            action.moveMagnitude = action.moveMagnitude > 0 ? action.moveMagnitude : gridSize;
+        public virtual void MoveRight(
+            float? moveMagnitude = null,
+            string objectId = "",
+            float maxAgentsDistance = -1f,
+            bool forceAction = false,
+            bool manualInteract = false,
+            bool allowAgentsToIntersect = false
+        ) {
+            if (!moveMagnitude.HasValue) {
+                moveMagnitude = gridSize;
+            } else if (moveMagnitude < 0f) {
+                throw new InvalidOperationException("moveMagnitude but be > 0");
+            }
```

Comment:
Should an invalid operation exception occur when `moveMagnitude == 0` (i.e., not moving the agent)?

Reply:
Good point, done.